Gems/Atom/RHI/DX12/Code/Source/RHI/AsyncUploadQueue.cpp

```diff
@@ -507,12 +507,19 @@ namespace AZ
         {
             AZ_PROFILE_SCOPE(RHI, "AsyncUploadQueue: ProcessCallbacks");
             AZStd::lock_guard<AZStd::mutex> lock(m_callbackMutex);
+            uint64_t completedValue = m_uploadFence.GetCompletedValue();
+            fenceValue = AZStd::min(completedValue, fenceValue);
             while (m_callbacks.size() > 0 && m_callbacks.front().second <= fenceValue)
             {
                 AZStd::function<void()> callback = m_callbacks.front().first;
                 m_callbacks.pop();
                 callback();
             }
+
+            if (!m_callbacks.empty())
```

Comment:
Can you add a comment why we are re-queueing the current function. Any specific use case?

Reply:
Added